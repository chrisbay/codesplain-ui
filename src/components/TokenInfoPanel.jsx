import React, { PropTypes } from 'react';
  import { Card, CardTitle, CardText } from 'material-ui/Card';
import { Tabs, Tab } from 'material-ui/Tabs';

import AnnotationDisplay from './AnnotationDisplay';
import AnnotationEditor from './AnnotationEditor';

const AnnotationSection = ({ displayStatus, displayProps }) => {
  switch (displayStatus) {
  case 'none': {
    return (
      <CardText>Click on a line to add an annotation or display one</CardText>
    );
  }
  case 'display': {
    return (
      <AnnotationDisplay
        {...displayProps}
      />
    );
  }
  case 'create': {
    return (
      <AnnotationEditor
        {...displayProps}
      />
    );
  }
  default: {
    return null;
  }
  }
}

const TokenInfoPanel = ({ displayStatus, displayProps, saveAnnotationCallback }) => {
  return (
    <Card>
      <CardTitle title="Info Panel" />
      <CardText>General info about the selected token goes here</CardText>
      <Tabs>
        <Tab label="Annotation">
          <AnnotationSection
            displayStatus={displayStatus}
            displayProps={displayProps}
          />
        </Tab>
        <Tab label="SyntaxDB">
          <CardText>Here is some info from SyntaxDB about the token.</CardText>
        </Tab>
      </Tabs>
    </Card>
  );
};

TokenInfoPanel.propTypes = {
  displayStatus: PropTypes.string.isRequired,
  displayProps: PropTypes.object.isRequired,
};

export default TokenInfoPanel;
