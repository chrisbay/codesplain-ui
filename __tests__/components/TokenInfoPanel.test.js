import React from 'react'
import renderer from 'react-test-renderer';

import TokenInfoPanel from '../../src/components/TokenInfoPanel'

jest.mock('material-ui/Card')
jest.mock('material-ui/Tabs')

describe('<TokenInfoPanel />', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <TokenInfoPanel />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
