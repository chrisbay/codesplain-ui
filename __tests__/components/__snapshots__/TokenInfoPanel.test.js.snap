exports[`<TokenInfoPanel /> matches snapshot 1`] = `null`;
